File: src/ts-packages/sdk/src/tool.ts
import type { ServerTool, ToolDefinition } from "./types.js";

export function tool(def: ToolDefinition): ServerTool {
  return {
    name: def.name,
    description: def.description,
    parameters: def.parameters,
    ...(def.metadata && { metadata: def.metadata }),
    fields: {
      name: def.name,
      description: def.description,
      inputSchema: JSON.stringify(def.parameters),
    },
  };
}

File: src/py-packages/sdk/src/agentified/tool.py
from __future__ import annotations

import json
from typing import Any

from .models import ServerTool, ServerToolFields


def tool(
    *,
    name: str,
    description: str,
    parameters: dict[str, Any],
    metadata: dict[str, Any] | None = None,
    always_include: bool | None = None,
) -> ServerTool:
    return ServerTool(
        name=name,
        description=description,
        parameters=parameters,
        **({"metadata": metadata} if metadata else {}),
        **({"always_include": always_include} if always_include is not None else {}),
        fields=ServerToolFields(
            name=name,
            description=description,
            input_schema=json.dumps(parameters),
        ),
    )
